design/release_automation.md

```diff
@@ -0,0 +1,87 @@
+# Release Automation
+
+This document is the design document for automating our release process for the Python CLI/SDK project. We have a requirement to support four different platforms (RHEL 7/8, macOS, and Windows) and need to create automated flows for each of them. 
+
+#### Milestone 1
+
+1. Spin up 4 machines/containers, pack the CLI executable, and add to artifacts of the build
+   -  The artifacts will be used for testing
+
+#### Milestone 2
+
+1. Run the integration tests on those machines as part of our pipeline. We want to run the tests on new machines that don't have Python or any other dependencies to ensure our tests accurately represent user environments.
```

Comment:
Very common for linux systems to have an out of date 'system' python. May be worth testing this case also.

Reply:
This, to the extent that LTS distros will still be running Python 2 (RHEL will be the biggest offender here) and any LTS distro will be eventually running EoL Python well before it goes EoL itself.